sdk/cpp/core/src/gnmi_provider.hpp

```diff
@@ -18,45 +18,40 @@
 #ifndef _GNMI_PROVIDER_H_
 #define _GNMI_PROVIDER_H_
 
-#include <memory>
-#include <string>
-#include <libgnmi/gnmi.grpc.pb.h>
-#include <libgnmi/gnmi.pb.h>
 #include "gnmi_client.hpp"
+#include "ietf_parser.hpp"
 #include "path_api.hpp"
 
-namespace ydk {
-
-	class gNMIClient;
-
-	class gNMIServiceProvider : public path::ServiceProvider {
-	public:
-			gNMIServiceProvider(path::Repository & repo,
-								   std::string address,
-								   std::string username,
-								   std::string password);
-			gNMIServiceProvider(std::string address,
-								   std::string username,
-								   std::string password);
-			~gNMIServiceProvider();
-			path::RootSchemaNode& get_root_schema() const;
-			std::shared_ptr<path::DataNode> invoke(path::Rpc& rpc) const;
-			EncodingFormat get_encoding() const;
-			void print_paths(ydk::path::SchemaNode& sn) const;
-			void print_data_paths(ydk::path::DataNode& dn) const;
-			void print_root_paths(ydk::path::RootSchemaNode& rsn) const;
-
-	private:
-		    std::shared_ptr<path::DataNode> handle_read(path::Rpc& rpc) const;
-			void initialize(path::Repository& repo);
-			std::string execute_payload(const std::string & payload) const;
-
-	private:
-			std::unique_ptr<gNMIClient> client;
-			//std::unique_ptr<path::ModelProvider> model_provider;
-			std::shared_ptr<ydk::path::RootSchemaNode> root_schema;
-			std::vector<std::string> server_capabilities;
-
-	};
+namespace ydk 
+{
+    class gNMIClient;
+
+    class gNMIServiceProvider : public path::ServiceProvider 
+    {
+    public:
+        typedef struct SecureChannelArguments
+        {
+        std::shared_ptr<grpc::ChannelCredentials> channel_creds;
+        grpc::ChannelArguments args;
+            
+        } SecureChannelArguments;
+
+        gNMIServiceProvider(path::Repository & repo, std::string address);
+        gNMIServiceProvider(std::string address);
+        ~gNMIServiceProvider();
+
+    private:
+        std::shared_ptr<path::DataNode> invoke(path::Rpc& rpc) const;
+        path::RootSchemaNode& get_root_schema() const;
```

Comment:
Were `invoke` and `get_root_schema` meant to be made public? These are being overridden here from the base `ServiceProvider` class where they are public methods